packages/resolve-query/src/index.js

```diff
@@ -1,44 +1,52 @@
+import 'regenerator-runtime/runtime';
+
 function updateState(projection, event, state) {
     return projection.eventHandlers[event.type](state, event);
 }
 
-const executor = ({ store, bus, projection }) => {
+function getExecutor({ eventStore, projection }) {
     const eventTypes = Object.keys(projection.eventHandlers);
     let state = projection.initialState || {};
+    let error = null;
+
+    const eventStream = eventStore.getStreamByEventTypes(eventTypes);
+
+    eventStream.on('readable', () => {
+        let event;
+        // eslint-disable-next-line no-cond-assign
+        while (null !== (event = eventStream.read())) {
+            try {
+                state = updateState(projection, event, state);
+            } catch (err) {
+                error = err;
+            }
+        }
+    });
 
-    const handler = event => (state = updateState(projection, event, state));
-
-    let result = null;
-    return () => {
-        result =
-            result ||
-            store.loadEventsByTypes(eventTypes, handler).then(() => {
-                bus.onEvent(eventTypes, handler);
-            });
+    eventStream.on('error', err => (error = err));
 
-        return result.then(() => state);
+    return async () => {
```

Comment:
Yes, it can be easily refactored, but we consider to not change API (After our common meeting)
If do it, it can possible affect API in future.